src/AzureOpenAIProxy.Management/Services/ModelService.cs

```diff
@@ -8,20 +8,46 @@
 
 namespace AzureOpenAIProxy.Management.Services;
 
-public class ModelService(IAuthService authService, AoaiProxyContext db, IConfiguration configuration) : IModelService
+public class ModelService() : IModelService, IDisposable
 {
-
     private const string PostgresEncryptionKey = "PostgresEncryptionKey";
-    private readonly DbConnection connection = db.Database.GetDbConnection();
+    private readonly IConfiguration configuration = null!;
+    private readonly IAuthService authService = null!;
+    private readonly AoaiProxyContext db = null!;
+    private readonly DbConnection conn = null!;
+
+    public ModelService(IAuthService authService, IDbContextFactory<AoaiProxyContext> dbContextFactory, IConfiguration configuration)
```

Comment:
This needs to be this way else you'd have to call CreateDbContext twice. The second time to create conn. This could be an issue as though unlikely, the second call could return a different DbContext given due to connection renewal for managed identity.

db = dbContextFactory.CreateDbContext();
conn = dbContextFactory.CreateDbContext().Database.GetDbConnection();

This is because you can't contruct the folowing in the primary constructor, so create a default constructor...

db = dbContextFactory.CreateDbContext();
conn = db.Database.GetDbConnection();